### PR TITLE
Implement command to export existing mail templates to disk

### PR DESCRIPTION
**Pull request by retiolum**
In my existing projects I often had the need to dump the existing mail templates to disk to simplify starting to use the plugin. The new command does exactly that.

<a href="https://gitpod.io/#https://github.com/FriendsOfShopware/FroshPlatformTemplateMail/pull/36"></a>

### CODE REVIEW

File: src/Command/ExportCommand.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            ->setName('frosh:template-mail:export')
+            ->setDescription('Export mail templates from database to file system')
+            ->addArgument('directory', InputArgument::REQUIRED, 'Target directory')
+            ->addOption('dry-run', null, InputOption::VALUE_NONE, 'Dry run (only show what is being done)');
+    }
+
+    /**
+     * @inerhitDoc
+     */
+    public function execute(InputInterface $input, OutputInterface $output)
```
Comment (shyim) [inline src/Command/ExportCommand.php:63]:
```suggestion
    public function execute(InputInterface $input, OutputInterface $output): int
```

File: src/Command/ExportCommand.php [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        $this->databaseConnection = $databaseConnection;
+    }
+
+    /**
+     * @inerhitDoc
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('frosh:template-mail:export')
```
Comment (shyim) [inline src/Command/ExportCommand.php:54]: This should be in a static property `$defaultName` to have a lazy command

File: src/Command/AbstractCommand.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Base class for all commands
+ */
+abstract class AbstractCommand extends \Symfony\Component\Console\Command\Command
```
Comment (shyim) [inline src/Command/AbstractCommand.php:15]: Do we really need this abstraction?

File: src/Command/ExportCommand.php [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+        // Handle directory.
+        $this->directory = $input->getArgument('directory');
+        if (!$isDryRun && !$this->initializeFilesystem()) {
+            return 1;
+        }
+
+        // Fetch mail templates.
+        $query = <<<EOT
+SELECT mail_template_translation.*, mail_template_type.technical_name, locale.code AS locale
```
Comment (shyim) [inline src/Command/ExportCommand.php:79]: We should select only the specific columns we need. Selecting a wildcard can be problematic when someone adds a similar column  to the first table

**Comment by shyim** [comment]
Looks like a good change. Will do the comments by own

**Comment by retiolum** [comment]
Thanks for the merge! The code was developed as part of a customer project and I was too busy to do the changes.

**Pull request merged**